### PR TITLE
[ALLUXIO-3204] Allow users to set app id

### PR DESCRIPTION
**Pull request by calvinjia**
https://alluxio.atlassian.net/browse/ALLUXIO-3204

### CODE REVIEW

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 1]
```diff
@@ -2406,6 +2406,13 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .setScope(Scope.CLIENT)
           .build();
+  public static final PropertyKey USER_APP_ID =
+      new Builder(Name.USER_APP_ID)
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/PropertyKey.java]: Keep in mind that multiple clients sharing the same configuration file would have the same app id, which would defeat the purpose of having a client id.

Comment (calvinjia) [reply core/common/src/main/java/alluxio/PropertyKey.java]: Right, if the user sets the app id manually, they should make sure they set it uniquely per logical application (for example in a Spark job).

Comment (yupeng9) [reply core/common/src/main/java/alluxio/PropertyKey.java]: Right. Can you call this out explicitly? Other than it, this PR looks good to me.

Comment (calvinjia) [reply core/common/src/main/java/alluxio/PropertyKey.java]: Updated, thanks.

**Pull request merged**